core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -874,20 +843,70 @@ object JsonAST {
     }
   }
 
+  sealed trait AppendContainer[T] {
```

Comment:
Just to make sure, did you try `StringBuilder` or `StringBuffer`? One of the two is synchronized and therefore slower. In particular, a quick peek at Scala's `StringBuilder` indicates it uses the [Java one](https://github.com/scala/scala/blob/v2.10.4/src/library/scala/collection/mutable/StringBuilder.scala#L28) [under the covers](https://github.com/scala/scala/blob/v2.10.4/src/library/scala/collection/mutable/StringBuilder.scala#L196-L199) anyway.